src/main/java/jenkins/plugins/openstack/compute/SimplifiedServer.java

```diff
@@ -0,0 +1,75 @@
+package jenkins.plugins.openstack.compute;
+
+import jenkins.plugins.openstack.compute.internal.DestroyMachine;
+import jenkins.plugins.openstack.compute.internal.Openstack;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+import org.openstack4j.model.compute.Server;
+
+import java.io.Serializable;
+
+public class SimplifiedServer implements Serializable{
+
+    private Server srv = null;
+    private String cloud = "";
+    private String template = "";
+    private String scope = "";
+
+    public SimplifiedServer(String cloud, String template, String scope) {
+        this.template = template;
+        this.cloud = cloud;
+        this.scope = scope;
+
+        ServerScope serverscope = ServerScope.parse(scope);
+        JCloudsCloud jcl = JCloudsCloud.getByName(cloud);
+        JCloudsSlaveTemplate t = jcl.getTemplate(template);
+        if (t != null) {
+            this.srv = t.provision(jcl, serverscope);
+        } else {
+            throw new IllegalArgumentException("Invalid template: " + template);
+        }
+    }
+
+    @Whitelisted
+    public void destroy() {
+        if (srv != null) {
+            DestroyMachine dm = new DestroyMachine(this.cloud, srv.getId());
+            dm.dispose();
+            srv = null;
+        } else {
+            //Cant destroy what isn't created
```

Comment:
Yes, all of these null checks cover invoking methods over a previously destroyed instance